.github/workflows/e2e-tests.yml

```
@@ -65,10 +65,10 @@ jobs:
           make --directory e2e-tests tests upgrade.bats
         shell: bash
         env:
-          CLUSTER_CONTEXT: k3d-${{ github.repository_owner }}-ghactions-cluster #TODO get context from setup-kubewarden-cluster-action
+          CLUSTER_NAME: ${{ github.repository_owner }}-ghactions-cluster
       - name: "Uninstall Kuberwarden"
         run: |
           # TODO - share release with the create-kubewarden-cluster action
-          helm uninstall -n kubewarden kubewarden-controller kubewarden-crds kubewarden-defaults
+          helm uninstall -n kubewarden kubewarden-defaults kubewarden-controller kubewarden-crds
```

```suggestion
          helm uninstall --wait -n kubewarden kubewarden-defaults
          helm uninstall --wait -n kubewarden kubewarden-controller
          helm uninstall --wait -n kubewarden kubewarden-crds
```

I'm not sure if providing a list of helm releases to Helm means Helm ensures that the uninstall happens in that specific order.

Also, the wait may be needed, if not, there can be problems. For example, both the `helm uninstall kubewarden-defaults` and the pre-delete-hook job of kubewarden-controller that does a `kubectl delete PolicyServer` may happen at the same time, failing somehow.
Or the controller may be half-way uninstalling (running the pre-delete-hook, and then removing the kubewarden-controller deployment), and then the crds are removed, which will break things.

---

Good catch, thanks! It seems helm uninstall happens [in order](https://github.com/helm/helm/blob/2062a59566056f4900c139b6425ceb66df893177/cmd/helm/uninstall.go#L54).
But `--wait` parameter is useful, kubectl delete defaults to `--wait`, but it seems helm does not.